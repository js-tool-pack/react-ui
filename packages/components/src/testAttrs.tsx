import { fireEvent, render } from '@testing-library/react';
import type { PropsBase } from '@pkg/shared';
import React from 'react';

export function testAttrs<
  T extends { attrs?: Partial<React.HTMLAttributes<HTMLElement>> },
>(component: React.FC<T>, displayName?: string) {
  const name = displayName || component.displayName || component.name;
  test(name ? `${name} attrs` : `attrs`, () => {
    const onClick = jest.fn();

    const FC = component as React.FC<PropsBase>;
    const { container } = render(
      <FC
        attrs={{
          style: { width: '20px' },
          className: 'foo',
          role: 'note',
          onClick,
        }}
      />,
    );

    expect(container.firstChild).toHaveClass('foo');
    expect(container.firstChild).toHaveAttribute('role', 'note');
    expect(container.firstChild).toHaveStyle({ width: '20px' });

    expect(onClick).not.toBeCalled();
    fireEvent.click(container.firstChild!);
    expect(onClick).toBeCalled();
  });
}
