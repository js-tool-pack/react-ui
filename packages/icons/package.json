{
  "version": "0.0.2-beta.3",
  "types": "dist/index.d.ts",
  "main": "dist/index.js",
  "private": true,
  "scripts": {
    "rm:dist": "rimraf ./dist",
    "build:this": "tsc -p tsconfig.json",
    "build": "pnpm rm:dist && pnpm build:this",
    "check:ts": "tsc --noEmit -p tsconfig.json"
  },
  "devDependencies": {}
}
