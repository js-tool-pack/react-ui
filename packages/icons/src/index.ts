export * from './close.icon';
export * from './loading.icon';
export * from './circle-info.icon';
export * from './circle-info-fill.icon';
export * from './circle-success.icon';
export * from './circle-success-fill.icon';
export * from './circle-warning.icon';
export * from './circle-warning-fill.icon';
export * from './circle-close.icon';
export * from './circle-close-fill.icon';
export * from './up.icon';
export * from './right.icon';
export * from './down.icon';
export * from './left.icon';
export * from './selected.icon';
export * from './eye.icon';
export * from './eye-invisible.icon';
export * from './download.icon';
export * from './reset.icon';
export * from './zoom-in.icon';
export * from './zoom-out.icon';
export * from './rotate-left.icon';
export * from './rotate-right.icon';
export * from './flip-h.icon';
export * from './flip-v.icon';
export * from './empty.icon';
export * from './left-double.icon';
export * from './right-double.icon';
