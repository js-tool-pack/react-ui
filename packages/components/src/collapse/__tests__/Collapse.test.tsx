import { render } from '@testing-library/react';
import { testAttrs } from '~/testAttrs';
import { Left } from '@pkg/icons';
import { Collapse } from '..';

describe('Collapse', () => {
  testAttrs(Collapse);

  test('basic', () => {
    expect(
      render(
        <Collapse title={'第十四章'}>
          视之不见名曰夷，听之不闻名曰希，搏之不得名曰微。
        </Collapse>,
      ).container.firstChild,
    ).toMatchSnapshot();
  });

  test('size', () => {
    expect(
      render(
        <Collapse>视之不见名曰夷，听之不闻名曰希，搏之不得名曰微。</Collapse>,
      ).container.firstChild,
    ).toHaveClass('t--size-m');
    expect(
      render(
        <Collapse size="medium">
          视之不见名曰夷，听之不闻名曰希，搏之不得名曰微。
        </Collapse>,
      ).container.firstChild,
    ).toHaveClass('t--size-m');
    expect(
      render(
        <Collapse size="small">
          视之不见名曰夷，听之不闻名曰希，搏之不得名曰微。
        </Collapse>,
      ).container.firstChild,
    ).toHaveClass('t--size-sm');
    expect(
      render(
        <Collapse size="large">
          视之不见名曰夷，听之不闻名曰希，搏之不得名曰微。
        </Collapse>,
      ).container.firstChild,
    ).toHaveClass('t--size-lg');
  });

  test('disabled', () => {
    expect(
      render(
        <Collapse title={'第十四章'} disabled>
          视之不见名曰夷，听之不闻名曰希，搏之不得名曰微。
        </Collapse>,
      ).container.firstChild,
    ).toHaveClass('t-collapse--disabled');
  });

  test('expanded', () => {
    expect(
      render(
        <Collapse title={'第十四章'} expanded>
          视之不见名曰夷，听之不闻名曰希，搏之不得名曰微。
        </Collapse>,
      ).container.firstChild,
    ).toHaveClass('t-collapse--active');
  });

  test('extra', () => {
    expect(
      render(
        <Collapse title={'第十四章'} extra="《道德经》">
          视之不见名曰夷，听之不闻名曰希，搏之不得名曰微。
        </Collapse>,
      ).container.firstChild,
    ).toMatchSnapshot();
  });

  test('icon', () => {
    expect(
      render(
        <Collapse icon={() => <Left />} title={'第十四章'}>
          视之不见名曰夷，听之不闻名曰希，搏之不得名曰微。
        </Collapse>,
      ).container.firstChild,
    ).toMatchSnapshot();
  });

  test('icon-placement', () => {
    expect(
      render(
        <Collapse icon={() => <Left />} iconPlacement="end" title={'第十四章'}>
          视之不见名曰夷，听之不闻名曰希，搏之不得名曰微。
        </Collapse>,
      ).container.firstChild,
    ).toMatchSnapshot();
  });
});
