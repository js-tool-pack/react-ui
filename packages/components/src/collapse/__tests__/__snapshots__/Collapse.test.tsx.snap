// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse basic 1`] = `
<section
  class="t-collapse t--size-m"
  role="tab"
>
  <div
    class="t-space t-collapse__header"
    style="--t-space-fill-ratio: 100%; gap: 12px;"
  >
    <i
      class="t-icon t-collapse__icon"
      role="img"
    >
      <svg
        viewBox="0 0 1024 1024"
      >
        <path
          d="M636.91389854 512.22982846L217.86080278 92.94690423l81.32081573-81.20590149 500.25899726 500.52713049L299.14331376 1012.25899726l-81.20590149-81.282511z"
          fill="currentColor"
        />
      </svg>
    </i>
    <div
      class="t-collapse__title"
    >
      第十四章
    </div>
  </div>
</section>
`;

exports[`Collapse extra 1`] = `
<section
  class="t-collapse t--size-m"
  role="tab"
>
  <div
    class="t-space t-collapse__header"
    style="--t-space-fill-ratio: 100%; gap: 12px;"
  >
    <i
      class="t-icon t-collapse__icon"
      role="img"
    >
      <svg
        viewBox="0 0 1024 1024"
      >
        <path
          d="M636.91389854 512.22982846L217.86080278 92.94690423l81.32081573-81.20590149 500.25899726 500.52713049L299.14331376 1012.25899726l-81.20590149-81.282511z"
          fill="currentColor"
        />
      </svg>
    </i>
    <div
      class="t-collapse__title"
    >
      第十四章
    </div>
    《道德经》
  </div>
</section>
`;

exports[`Collapse icon 1`] = `
<section
  class="t-collapse t--size-m"
  role="tab"
>
  <div
    class="t-space t-collapse__header"
    style="--t-space-fill-ratio: 100%; gap: 12px;"
  >
    <svg
      viewBox="0 0 1024 1024"
    >
      <path
        d="M808.12630286 92.94690423l-81.20590149-81.20590149-500.29730201 500.52713049 500.25899727 500.02916876 81.24420623-81.28251096-419.01479101-418.78496257z"
        fill="currentColor"
      />
    </svg>
    <div
      class="t-collapse__title"
    >
      第十四章
    </div>
  </div>
</section>
`;

exports[`Collapse icon-placement 1`] = `
<section
  class="t-collapse t--size-m"
  role="tab"
>
  <div
    class="t-space t-collapse__header"
    style="--t-space-fill-ratio: 100%; gap: 12px;"
  >
    <div
      class="t-collapse__title"
    >
      第十四章
    </div>
    <svg
      viewBox="0 0 1024 1024"
    >
      <path
        d="M808.12630286 92.94690423l-81.20590149-81.20590149-500.29730201 500.52713049 500.25899727 500.02916876 81.24420623-81.28251096-419.01479101-418.78496257z"
        fill="currentColor"
      />
    </svg>
  </div>
</section>
`;
